UnityProject/Assets/Scripts/Managers/PoolManager.cs

```diff
@@ -155,132 +260,54 @@ public static GameObject NetworkClone(GameObject toClone, Vector3? position = nu
-	/// be surprised if it doesn't as there are LOTS of prefabs in the game which all have unique behavior for how they should spawn. If you are trying
-	/// to instantiate something and it isn't properly setting itself up, check to make sure each component that needs to set something up has
-	/// properly implemented a OnSpawnedServer and OnSpawnedClient method.</param>
-	/// <param name="position">world position to appear at. Defaults to HiddenPos (hidden / invisible)</param>
-	/// <param name="rotation">rotation to spawn with, defaults to Quaternion.identity</param>
-	/// <param name="parent">Parent to spawn under, defaults to no parent. Most things
-	/// should always be spawned under the Objects transform in their matrix. Many objects (due to RegisterTile)
-	/// usually take care of properly parenting themselves when spawned so in many cases you can leave it null.</param>
-	/// <returns>the newly created GameObject</returns>
+	/// <returns>the re-created object</returns>
 	[Server]
-	public static GameObject PoolNetworkInstantiate(String prefabName, Vector3? position = null, Transform parent = null, Quaternion? rotation = null)
+	public static GameObject PoolNetworkTestDestroyInstantiate(GameObject target)
 	{
-		GameObject prefab = GetPrefabByName(prefabName);
-		if (prefab == null)
+		if (!IsInstanceInit())
 		{
-			Logger.LogErrorFormat("Attempted to spawn prefab with name {0} which is either not an actual prefab name or" +
-			                " is a prefab which is not spawnable. Request to spawn will be ignored.", Category.ItemSpawn, prefabName);
 			return null;
 		}
 
-		return PoolNetworkInstantiate(prefab, position, parent, rotation);
-	}
-
-	/// <summary>
-	/// Spawn the item locally without syncing it over the network. OnSpawnedServer and OnSpawnedClient hooks will
-	/// not be called as this is only used for client side things.
-	/// </summary>
-	/// <param name="prefab">Prefab to spawn an instance of. </param>
-	/// <param name="position">world position to appear at. Defaults to HiddenPos (hidden / invisible)</param>
-	/// <param name="rotation">rotation to spawn with, defaults to Quaternion.identity</param>
-	/// <param name="parent">Parent to spawn under, defaults to no parent. Most things
-	/// should always be spawned under the Objects transform in their matrix. Many objects (due to RegisterTile)
-	/// usually take care of properly parenting themselves when spawned so in many cases you can leave it null.</param>
-	/// <returns>the newly created GameObject</returns>
-	public static GameObject PoolClientInstantiate(GameObject prefab, Vector3? position = null, Transform parent = null, Quaternion? rotation = null)
-	{
-		if (!IsInstanceInit())
+		var objBehavior = target.GetComponent<ObjectBehaviour>();
+		if (objBehavior == null)
 		{
-			return null;
+			Logger.LogErrorFormat("{0} has no ObjectBehavior thus cannot be pooled.", Category.ItemSpawn, target.name);
```

Comment:
```suggestion
			Logger.LogErrorFormat("{0} has no ObjectBehavior thus cannot be pooled.", Category.ItemSpawn, target.name);
			return null;
```
Tried to respawn something that isn't poolable and got an NRE that broke right click menu, I think it's best to avoid them if possible
<img width="405" alt="Scr 2019-09-08 в 10 26 42" src="https://user-images.githubusercontent.com/10403536/64485011-36683c80-d223-11e9-836a-1e5a5359ae0e.png">

```
NullReferenceException: Object reference not set to an instance of an object
PoolManager.PoolNetworkTestDestroyInstantiate (UnityEngine.GameObject target) (at Assets/Scripts/Managers/PoolManager.cs:282)
PlayerNetworkActions.CmdAdminRespawn (UnityEngine.GameObject toRespawn) (at Assets/Scripts/Player/PlayerNetworkActions.cs:750)
PlayerNetworkActions.CallCmdAdminRespawn (UnityEngine.GameObject toRespawn) (at <18a947598b4544ef8b655e0d3501b528>:0)
CustomNetTransform.AdminRespawn () (at Assets/Scripts/Transform/CustomNetTransform.cs:691)
RadialMenu.Update () (at Assets/Scripts/UI/Right click/RadialMenu.cs:225)
```

Reply:
I actually want this to work as well rather than doing nothing. I'll add logic so if it's not pooled, it will respawn using the non-pooled approach when you select that option.